src/org/jitsi/impl/neomedia/ProcessorUtility.java

```diff
@@ -136,18 +136,26 @@ else if (state == Processor.Realized)
 
         boolean interrupted = false;
 
-        // Wait until we get an event that confirms the
-        // success of the method, or a failure event.
-        // See StateListener inner class
-        while ((processor.getState() < state) && !failed)
+        Object stateLock = getStateLock();
+        synchronized (stateLock)
         {
-            Object stateLock = getStateLock();
-
-            synchronized (stateLock)
+            // Wait until we get an event that confirms the
+            // success of the method, or a failure event.
+            // See StateListener inner class
+            while ((processor.getState() < state) && !failed)
             {
                 try
                 {
-                    stateLock.wait();
+                    // don't wait forever, there is some other
+                    // problem where we wait on an already closed
+                    // processor and we nver leave this wait
+                    stateLock.wait(10000);
+
+                    if (processor.getState() < state)
+                    {
+                        // state not reached we consider failure
+                        setFailed(true);
```

Comment:
If the Processor passes through an intermediary state before the 10s elapse, we will trigger a failure here. I think we should execute this only if we've waited for a certain period (e.g. 10s)

Reply:
@bgrozev I just updated it adding new commit. Tell me if you want to move the change to the commit with the initial change and force push so we can keep both commit with both changes separate ... thanks.